jenkins/spark-tests.sh

```diff
@@ -159,6 +159,23 @@ export SCRIPT_PATH="$(pwd -P)"
 export TARGET_DIR="$SCRIPT_PATH/target"
 mkdir -p $TARGET_DIR
 
+run_iceberg_tests() {
+  ICEBERG_VERSION="0.13.1"
+  # get the major/minor version of Spark
+  ICEBERG_SPARK_VER=$(echo $SPARK_VER | cut -d. -f1,2)
+
+  # Iceberg does not support Spark 3.3+ yet
+  if [[ "$ICEBERG_SPARK_VER" < "3.3" ]]; then
+    SPARK_SUBMIT_FLAGS="$BASE_SPARK_SUBMIT_ARGS $SEQ_CONF \
+      --packages org.apache.iceberg:iceberg-spark-runtime-${ICEBERG_SPARK_VER}_2.12:${ICEBERG_VERSION} \
+      --conf spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions \
+      --conf spark.sql.catalog.spark_catalog=org.apache.iceberg.spark.SparkSessionCatalog \
+      --conf spark.sql.catalog.spark_catalog.type=hadoop \
+      --conf spark.sql.catalog.spark_catalog.warehouse=/tmp/spark-warehouse-$$" \
+      ./run_pyspark_from_build.sh -m iceberg --iceberg
+  fi
```

Comment:
nit, should this exit or print a message when it can't run iceberg but we've asked it to?